libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1106,11 +1106,22 @@ void CompilerUtils::convertType(
 			}
 			case DataLocation::CallData:
 			{
-				solUnimplementedAssert(!typeOnStack.isDynamicallyEncoded(), "");
-				m_context << Instruction::DUP1;
-				m_context << Instruction::CALLDATASIZE;
-				m_context << Instruction::SUB;
-				abiDecode({&targetType}, false);
+				if (typeOnStack.isDynamicallyEncoded())
+				{
+					solAssert(m_context.useABICoderV2(), "");
```

Comment:
I added assertion as compiler needs abi v2 to be enabled for dynamically encoded calldata structs. Otherwise it reports error. So, I didn't request feature to be enabled, just assert to be safe.

Reply:
Is this really working fine everywhere? What about this:
```
contract C {
  function f() public pure {
    uint[][] calldata x;
    assembly { x.offset = 0 x.length = 10 }
    uint[][] memory y = x;
  }
}
```

Reply:
(I mean the actual numbers should be changed for this to not revert, just asking about whether this passes the analysis phase)

Reply:
This is example is about arrays, but this case is for structs only. It is a part of big `switch`, and this case is actually when converting structs from calldata to memory only...